custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -560,13 +603,33 @@ def _predict_legacy_drum(self, data, model, **kwargs) -> RawPredictResponse:
             assert all(isinstance(label, str) for label in request_labels)
         extra_model_output = None
         if self._custom_hooks.get(CustomHooks.SCORE):
-            try:
-                # noinspection PyCallingNonCallable
-                predictions_df = self._custom_hooks.get(CustomHooks.SCORE)(data, model, **kwargs)
-            except Exception as exc:
-                self._log_and_raise_final_error(
-                    exc, "Model 'score' hook failed to make predictions."
-                )
+            if self._target_type == TargetType.TEXT_GENERATION and self._guard_pipeline:
+                try:
+                    predictions_df = self._guard_moderation_hooks[GUARD_SCORE_WRAPPER_NAME](
+                        data,
+                        model,
+                        self._guard_pipeline,
```

Comment:
Do you pass init hook back to wrapper hook?

Reply:
As mentioned below, init hook will return the pipeline object, which is passed to the score wrapper.